### DISCUSSION TITLE
Enabling redirect in One-Click SIWE causes infinite wallet signature loop on mobile

### DISCUSSION THREAD

**Discussion by Xirynx** [reown-com/appkit#2996] [Q&A]
I have tried to enable redirects after signing in with no success.

I am using NextJS + Next-Auth, with the One-Click SIWE solution (with Wagmi) provided by the docs.

I enable redirects in the siweConfig object inside the verifyMessage function as so:

``` ts
verifyMessage: async ({ message, signature }: SIWEVerifyMessageArgs) => {
    try {
      const success = await signIn('credentials', {
        message,
        redirect: true,
        signature,
        callbackUrl: '/my-protected-route'
      })

      return Boolean(success?.ok)
    } catch (error) {
      return false
    }
  },
```

This works fine on desktop with Metamask (have not tried other wallets). On mobile, after redirecting, the next-auth signout route is called for some reason, and then I am prompted to sign a message again. This is an infinite loop of sign message, redirect, sign message again etc...  This issue occurs with both Metamask and Trust Wallet. Curiously, this is not a problem with Uniswap Wallet and Metamask's built in browser.

I have resorted to just doing SIWE with Next-Auth the old fashioned way as a fix because debugging this has been near impossible for me.
Here is my new working onClick handler solution in my client-side sign in button for reference:

```ts

async function handleLogin() {
  const nonce = await getCsrfToken(); //  import from next-auth
  const messageParams = {
    domain: typeof window !== 'undefined' ? window.location.origin : '',
    uri: typeof window !== 'undefined' ? window.location.pathname : '',
    chains: [mainnet.id, sepolia.id, base.id],
    statement: 'Please sign with your account',
    nonce,
  };
  const message = formatMessage(messageParams, address); // import from @reown/appkit-siwe
  signMessage({ message }, { // import from wagmi
    onSuccess: async (data) => {
      signIn('credentials', { // import from next-auth
        message,
        callbackUrl: '/my-protected-route',
        redirect: true,
        signature: data,
      });
    },
    onError: (error) => {
      toast({
        variant: 'destructive',
        title: 'Uh oh!',
        description: error.message,
      });
    },
  });
}
```

**Comment by rkemergify** [comment]
Have you found a fix for this?
